sync/src/orphan_block_pool.rs

```diff
@@ -88,6 +90,31 @@ impl InnerPool {
                 .and_then(|blocks| blocks.get(hash).cloned())
         })
     }
+
+    /// cleanup expired blocks(epoch + EXPIRED_EPOCH < tip_epoch)
+    pub fn clean_expired_blocks(&mut self, tip_epoch: EpochNumber) -> Vec<packed::Byte32> {
+        let mut result = vec![];
+        let leaders: Vec<ParentHash> = self.leaders.iter().cloned().collect();
+        for hash in leaders.iter() {
+            if let Some(block) = self.get_block_by_parent(hash) {
+                if block.header().epoch().number() + EXPIRED_EPOCH < tip_epoch {
+                    // remove items in orphan pool and header map
+                    let descendants = self.remove_blocks_by_parent(&hash);
+                    for block in descendants {
+                        result.push(block.hash());
+                    }
+                }
+            }
+        }
+        result
+    }
+
+    /// get 1st block belongs to that parent
+    fn get_block_by_parent(&self, parent_hash: &packed::Byte32) -> Option<core::BlockView> {
```

Comment:
It is recommended to modify this function to 
```
fn need_clean(&self, tip_epoch) -> bool {
            self.blocks
            .get(parent_hash)
            .map(|map| map.iter().next().map(|(_, block)| block.header().epoch().number() + EXPIRED_EPOCH < tip_epoch)).unwrap_or_default()
}
```

Reply:
i agree with you.